### PR TITLE
Release candidate 1.0.0.rc1

### PR DESCRIPTION
**Pull request by mamoonraja**
This PR updates `ibm_cloud_sdk_core` to `1.0.0.rc1`, updates the authentication mechanism and regenerates the SDK using current API definitions.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -341,35 +337,26 @@ thr.join # Wait for the thread to finish before ending the program or running ot
...
-    version: "<version>",
-    url: "<service url>",
-    icp4d_token: "<your managed access token>",
-    authentication_type: "icp4d"
-  )
-  assistant.configure_http_client(disable_ssl_verification: true) # MAKE SURE SSL VERIFICATION IS DISABLED
+authenticator = IBMCloudSdkCore::CLoudPakForDataAuthenticator.new(
+  username: "<username>",
+  password: "<password>",
+  url: "<authentication url>",
```
Comment (mediumTaj) [inline README.md]: Does order matter here? I think we had url first. no problem if it doesnt matter

File: test/integration/test_personality_insights_v3.rb [hunk 2]
```diff
@@ -9,10 +9,13 @@ class PersonalityInsightsV3Test < Minitest::Test
...
     def before_all
+      authenticator = IBMCloudSdkCore::IamAuthenticator.new(
+        apikey: ENV["PERSONALITY_INSIGHTS_APIKEY"]
+      )
       @service = IBMWatson::PersonalityInsightsV3.new(
         version: "2017-10-13",
-        iam_apikey: ENV["PERSONALITY_INSIGHTS_APIKEY"],
-        url: ENV["PERSONALITY_INSIGHTS_URL"]
+        url: ENV["PERSONALITY_INSIGHTS_URL"],
+        authenticator: authenticator
```
Comment (mediumTaj) [inline test/integration/test_personality_insights_v3.rb]: do we need to pass in url here and in the other tests?

File: README.md (hunk 1 above)
Comment (mamoonraja) [reply README.md]: nope, doesn't matter

File: test/integration/test_personality_insights_v3.rb (hunk 2 above)
Comment (mamoonraja) [reply test/integration/test_personality_insights_v3.rb]: we don't need to but it doesn't hurt to pass it. So I am fine either ways.

**Pull request merged**